lib/spack/spack/variant.py

```diff
@@ -610,7 +610,9 @@ def substitute_abstract_variants(spec):
     for name, v in spec.variants.items():
         if name in spack.directives.reserved_names:
             continue
-        pkg_variant = spec.package_class.variants[name]
+        pkg_variant = spec.package_class.variants.get(name, None)
+        if not pkg_variant:
```

Comment:
What if the variant is False? Wouldn't it be safer to check if `pkg_variant is None`?